src/passes/Asyncify.cpp

```diff
@@ -671,6 +679,30 @@ class ModuleAnalyzer {
       module.removeFunction(name);
     }
 
+    // When propagateAddList is enabled, we should check a add-list before
+    // scannerpropagateBack so that callers of functions in add-list should also
+    // be instrumented.
+    if (propagateAddList) {
+      if (!addListInput.empty()) {
+        for (auto& func : module.functions) {
+          if (addList.match(func->name) && removeList.match(func->name)) {
+            Fatal() << func->name
+                    << " is found in the add-list and in the remove-list\n";
+          }
+
+          if (!func->imported() && addList.match(func->name)) {
+            auto& info = scanner.map[func.get()];
+            if (verbose && !info.canChangeState) {
+              std::cout << "[asyncify] " << func->name
+                        << " is in the add-list, add\n";
+            }
+            info.canChangeState = true;
+            info.addedFromList = true;
+          }
+        }
+      }
+    }
```

Comment:
This duplicated code could be moved to a helper function, or at least a lambda.

Reply:
done.